### DISCUSSION TITLE
Does Loopback 4 support multiple models with the same name (e.g. via use of namespaces)?

### DISCUSSION THREAD

**Discussion by dhurtrci** [loopbackio/loopback-next#7530] [Q&A]
Does Loopback 4 support multiple models with the same name (e.g. via use of namespaces)?

**Comment by achrinza** [comment]
It should be possible to bind Models with the same class name; Most of LoopBack 4's components (such as `DefaultCrudRepository`) don't try to resolve Models by themselves; Instead, they rely on the user resolving it themselves (e.g. via `@inject()`), hence does not make assumptions on _how_ the Models are bound to Context.

Though it is generally expected that Model Bindings are tagged with `RepositoryTags.MODEL`.

As for having multiple Models with auto-prefixing of the database table name, I do not believe this is possible. The Models need to have their `tableName` in `@model()` be manually set.
